File: examples/evaluators/http-evaluator.ts
/**
 * http-evaluator.ts — Example HTTP evaluator for optimize-anything.
 *
 * Scores candidates by word count proximity to a target.
 *
 * Start: bun run examples/evaluators/http-evaluator.ts
 * Test:  curl -X POST http://localhost:3456 \
 *          -H "Content-Type: application/json" \
 *          -d '{"candidate": "hello world"}'
 */

const TARGET_WORDS = 50;

const server = Bun.serve({
  port: 3456,
  async fetch(req) {
    if (req.method !== "POST") {
      return new Response("POST only", { status: 405 });
    }

    const body = (await req.json()) as { candidate?: string };
    const candidate = body.candidate ?? "";
    const words = candidate.split(/\s+/).filter(Boolean).length;
    const diff = Math.abs(words - TARGET_WORDS);
    const score = 1.0 / (1.0 + diff / 10.0);

    return Response.json({
      score,
      sideInfo: {
        wordCount: words,
        target: TARGET_WORDS,
        diff,
      },
    });
  },
});

console.error(`HTTP evaluator running on http://localhost:${server.port}`);
console.error(`Target: ${TARGET_WORDS} words`);

File: examples/evaluators/http_evaluator.py
#!/usr/bin/env python3
"""Example HTTP evaluator that scores by candidate length."""
import json
from http.server import HTTPServer, BaseHTTPRequestHandler

class Handler(BaseHTTPRequestHandler):
    def do_POST(self):
        body = self.rfile.read(int(self.headers.get("Content-Length", 0)))
        data = json.loads(body)
        candidate = data.get("candidate", "")
        score = min(len(candidate) / 100, 1.0)
        self.send_response(200)
        self.send_header("Content-Type", "application/json")
        self.end_headers()
        self.wfile.write(json.dumps({"score": score, "length": len(candidate)}).encode())

if __name__ == "__main__":
    HTTPServer(("localhost", 8000), Handler).serve_forever()
